src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
+  def emptyValue: A = empty
```

Comment:
It sounds like linux might deal with weird values in stat better than does OSX. But the value of having an empty value for each type is not because we expect it to be produced endlessly. It is primarily in being able to work with `K => V` and not `K => Option[V]`, with total functions instead of partial functions.